### DISCUSSION TITLE
Performance Optimizations

### DISCUSSION THREAD

**Discussion by sosiology** [visualize-admin/visualization-tool#996] [General]
Also See Google Doc: ENTER LINK

**Comment by ortnever** [comment]
Hi everybody, 

I did some Performance user tests from 28 February to 13 March, always under the same conditions:
- on int.visualize.admin.ch with data source PROD
- always using the same 3 datasets
- doing exactly the same operations.

The results are always very similar and the performance is good (1 to 4 sec. for the display of the preview, respectively of the graphs), except for a larger dataset (Stapfer-Surveys), for which it takes 11 to 14 seconds for the preview and 20 to 31 seconds for the display of the graph. Attached is the excel file (I could not put it in google doc).

The same tests performed on visualize PROD give the same results as on INT. 

If we consider these manual tests, I think that we could go to PROD with the current INT release. To be discussed with Fabian who has automated performance tests on his side. 

However, it would be a good thing to display a message warning the user that the display of larger datasets can take more than 30 seconds. 30 seconds is long enough for a user to leave the page. 

On 2. and 7. March I noticed a problem with the Stapfer-Schule dataset which may have indirectly something to do with performance?? The default graph was not working (Message: no data for the selected filter combination) and the filter dropdown list was empty, so it was impossible to make another selection and display any graph. Unfortunately I could not reproduce the problem. I onlyhave a print screen in the attached [excel.]

[Check_PerformanzVisualize_Benutzer.xlsx](https://github.com/visualize-admin/visualization-tool/files/10954685/Check_PerformanzVisualize_Benutzer.xlsx)

**Reply by adintegra** [reply]
Thanks for the comprehensive and systematic testing @ortnever – much appreciated! We will discuss in the meeting later today.

**Comment by ortnever** [comment]
By action "Start a visualization" for the cube "Stapfer survey", the screen ist completly white during the 10 first seconds.
![image](https://user-images.githubusercontent.com/79834090/224645844-6273f110-8ba9-4004-95b7-1ce1d7dbac6d.png)




**The user should at least see the "timer"** and eventually with a message like "Larger cubes may take longer than 30 seconds to be displayed"

![image](https://user-images.githubusercontent.com/79834090/224645451-05f950ba-48bf-4477-9a03-db8e3e395096.png)

**Comment by FabianCretton** [comment]
Here my tests, with an explanation on the first tab (but in french currently)

[Perf Visualize graphQL_Résultats 3 jours.xlsx](https://github.com/visualize-admin/visualization-tool/files/10968814/Perf.Visualize.graphQL_Resultats.3.jours.xlsx)
